`use strict` surrounded by backticks not correctly recognized by the strict rules being valid when quote backticks  rule is also enforced

## Detailed information about the problem
For some reason, `use strict` surrounded by backticks seems to be incorrectly flagged as not adhering to rules regarding `strict` being used (`"strict`, deprecated `"global-strict"`) when strings are also enforce being backticks `"quotes": [2, "backticks" ]`.

From all the JS environments I've used w/ the support of ES6 template strings (Chrome, Firefox, Babel, & I think Microsoft Edge as well), `use strict ` with backticks is correctly recognized as "use strict".

If I'm misunderstood about the implication of these rules, feel free to let me know! 

## Relevant rules
Here's the rules I'm using to get this behavior:
```js
 "quotes": [
     1,
    "backtick"
  ],

   "strict": [
      2,
      "global"
    ],

  "global-strict" :[
    0
  ],
```

## Other Relevant Information
- ESLint version `v0.24.1`

## Entire Config Object
```js

const ESLintConfig = {
    "rules": {
        "indent": [
            2,
            2,
        ],

        "dot-location":  [
          2,
          "object"
        ],

        "no-return-assign": [
          2
        ],

        "camelcase": [
          2,
          {
            properties: "always"
          },
        ],

        "no-inline-comments": [
          2,
        ],

        "max-nested-callbacks": [
          2,
          3,
        ],

        "linebreak-style": [
            2,
            "unix",
        ],

        "semi": [
            2,
            "always",
        ],

        "use-isnan": [
          2,
        ],

        "space-in-parens": [
          2,
          "always",
          {
            "exceptions": [ "empty", "[]" ],
          }
        ],

        "valid-jsdoc": [
          2,
        ],

        "no-mixed-spaces-and-tabs": [
          2,
        ],

        "quotes": [
          1,
          "backtick"
        ],

        "no-extra-semi": [
          2,
        ],

        "no-var": [
          1,
        ],

        "no-this-before-super": [
          2,
        ],

        "prefer-const": [
          1
        ],

        "computed-property-spacing": [
          2,
          "always",
        ],

        "object-shorthand": [
          2,
          "always",
        ],

        "constructor-super": [
          2,
        ],

        "padded-blocks": [
          2,
          "always",
        ],

        "space-before-function-paren": [
          2,
          "never",
        ],

        "no-multiple-empty-lines": [
          2,
          {
            max: 2,
          }
        ],

        "comma-spacing": [
          2,
          {
            before: false,
            after: true,
          }
        ],

        "no-spaced-func": [
          2
        ],

        "comma-style": [
          2,
          "first",
        ],

        "new-cap": [
          2,
        ],

        "no-trailing-spaces": [
          2,
          {
            skipBlankLines: true,
          }
        ],

        "spaced-comment": [
          2,
          "always",
          {
            exceptions: [ "-" ],
            markers: ["="],
          }
        ],

        "space-infix-ops": [
          2,
          {
            int32Hint: true,
          }
        ],

        "func-style": [
          2,
          "expression",
        ],

        "array-bracket-spacing": [
          2,
          "always",
          {
            "singleValue": false,
            "objectsInArrays": false,
            "arraysInArrays": false,
          }
        ],

        "quote-props": [
          2,
          "as-needed",
        ],

        "valid-typeof": [
          2,
        ],

        "no-control-regex": [
          2,
        ],

        "no-regex-spaces": [
          2,
        ],

        "no-empty": [
          2,
        ],

        "no-extra-boolean-cast": [
          2,
        ],

        "no-dupe-keys": [
          2,
        ],

        "no-negated-in-lhs": [
          2,
        ],

        "no-obj-calls": [
          2,
        ],

        "no-ex-assign": [
          2,
        ],

        "no-unreachable": [
          2,
        ],

        "no-constant-condition": [
          2,
        ],

        "no-sparse-arrays": [
          2,
        ],

        "no-unexpected-multiline": [
          2,
        ],

        "no-dupe-args": [
          2,
        ],

        "no-cond-assign": [
          2,
        ],

        "no-unused-vars": [
          1,
        ],

        "vars-on-top": [
          2,
        ],

        "no-new-wrappers": [
          2,
        ],

        "no-eval": [
          2,
        ],

        "no-multi-spaces": [
          2,
        ],

        "eqeqeq": [
          2,
          "smart"
        ],

        "no-script-url": [
          2,
        ],

        "no-throw-literal": [
          2,
        ],

        "comma-dangle": [
          2,
          "always-multiline",
        ],

        "strict": [
          2,
          "global"
        ],

        "global-strict" :[
          0
        ],

    },
    "env": {
        "es6": true,
        "browser": true,
        "mocha": true
    },

    "globals": {
      "dialogPolyfill": true,
      "$": true,
      "jQuery": true
    }
}

```

Yes, Templates cannot be directives. So...

## test.js

```js
/* eslint quotes: [2, "backtick"] */
"use strict";

var foo = `test`;
```

## Command & Actual

```
> node .\bin\eslint.js test.js --no-eslintrc --env es6

test.js
  2:1  error  Strings must use backtick  quotes

✖ 1 problem (1 error, 0 warnings)
```

## Expected

No errors.

--- a/lib/rules/quotes.js
+++ b/lib/rules/quotes.js
@@ -29,6 +29,8 @@ var QUOTE_SETTINGS = {
 
 var AVOID_ESCAPE = "avoid-escape";
 
+var FUNCTION_TYPE = /^(?:Arrow)?Function(?:Declaration|Expression)$/;
+
 //------------------------------------------------------------------------------
 // Rule Definition
 //------------------------------------------------------------------------------
@@ -56,6 +58,53 @@ module.exports = function(context) {
         return node.type.indexOf("JSX") === 0;
     }
 
+    /**
+     * Checks whether or not a given node is a directive.
+     * The directive is a `ExpressionStatement` which has only a string literal.
+     * @param {ASTNode} node - A node to check.
+     * @returns {boolean} Whether or not the node is a directive.
+     * @private
+     */
+    function isDirective(node) {
+        return (
+            node.type === "ExpressionStatement" &&
+            node.expression.type === "Literal" &&
+            typeof node.expression.value === "string"
+        );
+    }
+
+    /**
+     * Checks whether or not a given node is a part of directive prologues.
+     * See also: http://www.ecma-international.org/ecma-262/6.0/#sec-directive-prologues-and-the-use-strict-directive
+     * @param {ASTNode} node - A node to check.
+     * @returns {boolean} Whether or not the node is a part of directive prologues.
+     * @private
+     */
+    function isPartOfDirectivePrologue(node) {
+        if (!isDirective(node.parent)) {
+            return false;
+        }
+
+        var block = node.parent.parent;
+        if (block.type !== "Program" && (block.type !== "BlockStatement" || !FUNCTION_TYPE.test(block.parent.type))) {
+            return false;
+        }
+
+        // Check the node is at a prologue.
+        for (var i = 0; i < block.body.length; ++i) {
+            var statement = block.body[i];
+
+            if (statement === node.parent) {
+                return true;
+            }
+            if (!isDirective(statement)) {
+                break;
+            }
+        }
+
+        return false;
+    }
+
     return {
 
         "Literal": function(node) {
@@ -67,7 +116,7 @@ module.exports = function(context) {
                 isValid;
 
             if (settings && typeof val === "string") {
-                isValid = isJSXElement(node.parent) || isSurroundedBy(rawVal, settings.quote);
+                isValid = (quoteOption === "backtick" && isPartOfDirectivePrologue(node)) || isJSXElement(node.parent) || isSurroundedBy(rawVal, settings.quote);
 
                 if (!isValid && avoidEscape) {
                     isValid = isSurroundedBy(rawVal, settings.alternateQuote) && rawVal.indexOf(settings.quote) >= 0;

--- a/tests/lib/rules/quotes.js
+++ b/tests/lib/rules/quotes.js
@@ -34,7 +34,14 @@ eslintTester.addRuleTest("lib/rules/quotes", {
         { code: "var foo = <div id=\"foo\"></div>;", args: [1, "backtick"], ecmaFeatures: { jsx: true } },
         { code: "var foo = <div>Hello world</div>;", args: [1, "backtick"], ecmaFeatures: { jsx: true }},
         { code: "var foo = `backtick`;", args: [1, "single"], ecmaFeatures: { templateStrings: true }},
-        { code: "var foo = `backtick`;", args: [1, "double"], ecmaFeatures: { templateStrings: true }}
+        { code: "var foo = `backtick`;", args: [1, "double"], ecmaFeatures: { templateStrings: true }},
+
+        // `backtick` should not warn the directive prologues.
+        { code: "\"use strict\"; var foo = `backtick`;", options: ["backtick"], ecmaFeatures: { templateStrings: true }},
+        { code: "\"use strict\"; 'use strong'; \"use asm\"; var foo = `backtick`;", options: ["backtick"], ecmaFeatures: { templateStrings: true }},
+        { code: "function foo() { \"use strict\"; \"use strong\"; \"use asm\"; var foo = `backtick`; }", options: ["backtick"], ecmaFeatures: { templateStrings: true }},
+        { code: "(function() { 'use strict'; 'use strong'; 'use asm'; var foo = `backtick`; })();", options: ["backtick"], ecmaFeatures: { templateStrings: true }},
+        { code: "(() => { \"use strict\"; \"use strong\"; \"use asm\"; var foo = `backtick`; })();", options: ["backtick"], ecmaFeatures: { arrowFunctions: true, templateStrings: true }}
     ],
     invalid: [
         { code: "var foo = 'bar';",
@@ -62,6 +69,20 @@ eslintTester.addRuleTest("lib/rules/quotes", {
           errors: [{ message: "Strings must use backtick.", type: "Literal" }]},
         { code: "var foo = 'bar';",
           args: [1, "backtick", "avoid-escape"],
+          errors: [{ message: "Strings must use backtick.", type: "Literal" }]},
+
+        // below are not the directive prologues.
+        { code: "var foo = `backtick`; \"use strict\";",
+          options: ["backtick"],
+          ecmaFeatures: { templateStrings: true },
+          errors: [{ message: "Strings must use backtick.", type: "Literal" }]},
+        { code: "{ \"use strict\"; var foo = `backtick`; }",
+          options: ["backtick"],
+          ecmaFeatures: { templateStrings: true },
+          errors: [{ message: "Strings must use backtick.", type: "Literal" }]},
+        { code: "if (1) { \"use strict\"; var foo = `backtick`; }",
+          options: ["backtick"],
+          ecmaFeatures: { templateStrings: true },
           errors: [{ message: "Strings must use backtick.", type: "Literal" }]}
     ]
 });

